### ISSUE TITLE
Do not ignore inherited/included configuration on debug

### ISSUE THREAD

**Issue by EnergoStalin** [xmake-io/xmake-vscode#233]
### Is your feature request related to a problem? Please describe.

It's not taking in account included configs

I have structure like this

# xmake.lua
```lua
-- Top level xmake
includes("compiler.lua", "somemodule/xmake.lua")
```

# compiler.lua
```lua
-- Compiler with toolchain settings
add_rules("mode.release", "mode.debug")

add_cflags("-Wall", "-Werror", "-Wextra")
set_languages("c11")

add_toolchains("gcc")
```

# somemodule/xmake.lua
```lua
-- Module xmake
includes("test/xmake.lua")
```

# somemodule/test/xmake.lua
```lua
-- Test xmake
target("test")
	set_kind("binary")
```

When building target from the top level xmake it works perfectly because global configuration inherits top down but extension seems to ignore top level configs and going straight for target's xmake.lua even ignoring direct `includes("../../compiler")` in `somemodule/test/xmake.lua`

### Describe the solution you'd like

I'd like perform config evaluation like it's xmake does not ignoring top level settings for project

### Describe alternatives you've considered

I considered copy pasting compiler.lua directly into test/xmake.lua which obviously work. It Didn't work tho when including it like that `includes("../compiler.lua")` or via add_requireconfs.

### Additional context

In conclusion it works as expected only when contents of `compiler.lua` inserted directly into `somemodule/test/xmake.lua`

xmake v2.8.3+20230926
XMake v2.2.5

**Comment by waruqi** [comment]
>  but extension seems to ignore top level configs and going straight for target's xmake.lua even ignoring direct includes("../../compiler") in somemodule/test/xmake.lua

includes does not introduce the add_cflags/add_defines target APIs directly into the current xmake.lua, it only works for targets defined in compiler.lua and its sub-xmake.lua.

you can use function to wrap it if you want to set them to test/xmake.lua

compiler.lua

```lua
function set_flags_for_compiler()
    add_cflags("-Wall", "-Werror", "-Wextra")
    set_languages("c11")
    add_toolchains("gcc")
end
```

test/xmake.lua

```
includes("compiler.lua")

set_flags_for_compiler()

target("xxx")
```
